ambry-mysql/src/main/java/com/github/ambry/accountstats/AccountStatsMySqlStore.java

```diff
@@ -189,10 +208,17 @@ public void storeAggregatedStats(StatsSnapshot snapshot) throws SQLException {
         String containerIdKey = currContainerMapEntry.getKey();
         short containerId = Short.valueOf(containerIdKey.substring(2, containerIdKey.length() - 1));
         long currStorageUsage = currContainerMapEntry.getValue().getValue();
-        aggregatedaccountReportsDao.updateStorageUsage(accountId, containerId, currStorageUsage);
+        if (config.updateBatchSize >= 0) {
+          batch.addUpdateToBatch(accountId, containerId, currStorageUsage);
```

Comment:
If this throws SqlException, you want to catch it and call batch.rollback() since commit will not be called.

Reply:
I have different thoughts on this one. The way we structure the single write update is that 1) we have MySqlDataAccessor cache  the prepared statement 2) we have XXXXXDao class to supply values to prepared statement and then execute it, at the same time, catch the exception and rethrow it 3) we have higher functions to just call XXXXDao method to update mysql database. 
AccountStatsMySqlStore is a class at layer 3, it shouldn't deal with the exception. That's why I move the exception handing logic to BatchUpdater. In BatchUpdater, I will roll back when failing to add the parameter to batch.